gentle_web_align.py

```diff
@@ -0,0 +1,153 @@
+    gentle_web_align.py (--nested-directories=<main_directory>) [options]
+
+options:
+    -w <wav_pattern> --wav_pattern=<wav_pattern> Pattern of wav files to be aligned
+    -t <txt_pattern> --txt_pattern=<txt_pattern> Pattern of txt transcript files to be aligned (same name required)
+    --nested-directories=<main_directory>        Process every wav/txt file in the subfolders of the given folder
+    --server_addr=<server_addr>                  Server address that serves gentle. [default: localhost]
+    --port=<port>                                Server port that serves gentle. [default: 8567]
+    --max_unalign=<max_unalign>                  Maximum threshold for unalignment occurence (0.0 ~ 1.0) [default: 0.3] 
+    --skip-already-done                          Skips if there are preexisting .lab file
+    -h --help                                    show this help message and exit
+"""
+
+from docopt import docopt
+from glob import glob
+from tqdm import tqdm
+import os.path
+import requests
+import numpy as np
+
+def write_hts_label(labels, lab_path):
+    lab = ""
+    for s, e, l in labels:
+        s, e = float(s) * 1e7, float(e) * 1e7
+        s, e = int(s), int(e)
+        lab += "{} {} {}\n".format(s, e, l)
+    print(lab)
+    with open(lab_path, "w", encoding='utf-8') as f:
+        f.write(lab)
+
+
+def json2hts(data):
+    emit_bos = False
+    emit_eos = False
+
+    phone_start = 0
+    phone_end = None
+    labels = []
+    failure_count = 0
+    
+    for word in data["words"]:
+        case = word["case"]
+        if case != "success":
+            failure_count += 1 # instead of failing everything, 
+            #raise RuntimeError("Alignment failed")
+            continue
+        start = float(word["start"])
+        word_end = float(word["end"])
+
+        if not emit_bos:
+            labels.append((phone_start, start, "silB"))
+            emit_bos = True
+
+        phone_start = start
+        phone_end = None
+        for phone in word["phones"]:
+            ph = str(phone["phone"][:-2])
+            duration = float(phone["duration"])
+            phone_end = phone_start + duration
+            labels.append((phone_start, phone_end, ph))
+            phone_start += duration
+        assert np.allclose(phone_end, word_end)
+    if not emit_eos:
+        labels.append((phone_start, phone_end, "silE"))
+        emit_eos = True
+    unalign_ratio = float(failure_count) / len(data['words'])
+    return unalign_ratio, labels
+
+
+def gentle_request(wav_path,txt_path, server_addr, port, debug=False):
+    print('\n')
+    response = None
+    wav_name = os.path.basename(wav_path)
+    txt_name = os.path.basename(txt_path)
+    if os.path.splitext(wav_name)[0] != os.path.splitext(txt_name)[0]:
+        print(' [!] wav name and transcript name does not match - exiting...')
+        return response
+    with open(txt_path, 'r', encoding='utf-8-sig') as txt_file:
```

Comment:
Well, it was in my case (probably because I am currently mixing up with Windows (for running pyTorch) and Linux(for data preparation/alignment)), and I think that setting encoidng='utf-8-sig' when opening file is better for ensuring compatibility.